src/games/strategy/engine/framework/ui/NewGameChooserEntry.java

```diff
@@ -51,31 +59,34 @@ public NewGameChooserEntry(final URI uri)
   }
 
   public void fullyParseGameData() throws GameParseException {
+    // TODO: We should be setting this in the the constructor. At this point, you have to call methods in the
+    // correct order for things to work, and that is bads.
     m_data = null;
```

Comment:
yeah, this is an example of something a bit gnarly. All init is best done in a constructor, so "m_data" likely should be a final object. Otherwise as the comment mentions, method call ordering becomes very important, which makes objects difficult to use.